src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -95,6 +97,95 @@ CHIP_ERROR GeneralCommissioningAttrAccess::Read(const ConcreteReadAttributePath
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).version);
+    }
+    case TCMinRequiredVersion::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetRequirements(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).version);
+    }
+    case TCAcknowledgements::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).value);
+    }
+    case TCAcknowledgementsRequired::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+        TermsAndConditionsState termsAndConditionsState;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        err = termsAndConditionsProvider->CheckAcceptance(outTermsAndConditions, termsAndConditionsState);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        bool setTermsAndConditionsCallRequiredBeforeCommissioningCompleteSuccess =
+            termsAndConditionsState != TermsAndConditionsState::OK;
+        return aEncoder.Encode(setTermsAndConditionsCallRequiredBeforeCommissioningCompleteSuccess);
+    }
+    case TCUpdateDeadline::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        return aEncoder.EncodeNull();
```

Comment:
Is this feature currently under development?

Reply:
I intended to leave the default implementation to not support T&C updates, but I did want to allow the general-commissioning-server class to be generic. I'll add a `GetUpdateAcceptanceDeadline` function to the TermsAndConditionsProvider interface.